pkg/cmd/pulumi/new.go

```diff
@@ -658,6 +711,29 @@ func promptAndCreateStack(ctx context.Context, b backend.Backend, prompt promptF
 			return nil, err
 		}
 		s, err := stackInit(ctx, b, formattedStackName, root, setCurrent, secretsProvider)
+		// Check if stack already exists and allow the user to bypass it.
+		var e *backend.StackAlreadyExistsError
+		if errors.As(err, &e) {
+			accept, err := acceptStackAlreadyExists(stackName, opts)
+			if err != nil {
+				return nil, err
+			}
+			if !accept {
+				continue
+			}
+			stackRef, err := b.ParseStackReference(stackName)
```

Comment:
Should we print a warning or something here that we're ignoring the given `secretsProvider` option given to new because the stack already exists?

Reply:
Not allowing `pulumi new` to use an existing stack anymore.